sycl/doc/extensions/proposed/sycl_ext_oneapi_usm_malloc_properties.asciidoc

```diff
@@ -0,0 +1,1468 @@
+
+// APtr17 is of type annotated_ptr<int, decltype(properties{alignment<64>, usm_kind<sycl::usm::alloc::device>})>
+auto APtr17 = annotated_aligned_alloc_device<int>(N, q, 128, P11);
+
+// APtr18 is of type annotated_ptr<int, decltype(properties{alignment<64>, usm_kind<sycl::usm::alloc::device>})>
+auto APtr18 = annotated_aligned_alloc_device<int>(N, q, 16, P11);
+----
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification, Section 6.3.3 "Feature test macros". Therefore, an
+implementation supporting this extension must predefine the macro
+`SYCL_EXT_ONEAPI_USM_MALLOC_PROPERTIES` to one of the values defined in the table below.
+Applications can test for the existence of this macro to determine if the
+implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features
+the implementation supports.
+
+[%header,cols="1,5"]
+|===
+|Value |Description
+|1     |Initial extension version
+|===
+
+=== Orientation
+
+The core SYCL specification lists eight functions in each of the following four tables:
+
+- Table 103 "USM Device Memory Allocation Functions", of Section 4.8.3.2 "Device allocation functions" +
+- Table 104 "USM Host Memory Allocation Functions", of Section 4.8.3.3 "Host allocation functions" +
+- Table 105 "USM Shared Memory Allocation Functions", of Section 4.8.3.4 "Shared allocation functions" +
+- Table 106 "USM Parameterized Allocation Functions", of Section 4.8.3.5 "Parameterized allocation functions"
+
+This extension introduces a new function for each function listed in the above tables of the core SYCL specification.
+This extension also adds USM memory allocation functions with `properties` support that require the `sycl::ext::oneapi::experimental::usm_kind` property,
+and these do not correspond to any functions in the core SYCL specification.
+All USM memory allocation functions introduced by this extension are listed explicitly in Section <<section.usm.allocs>>.
+
+[NOTE]
+====
+The USM memory allocation functions defined in the core SYCL specification can be used in the same program as the USM memory allocation functions with `properties` support defined in this extension.
+The new functions are distinguished by their names having the `annotated_` prefix.
+====
+
+[[section.usm.allocs]]
+=== USM Memory Allocation Functions with properties Support
+
+The following five tables list all functions introduced by this extension.
+
+TODO: How is propertyListB going to be inferred? Do we need support for getting the type of a properties list with only the compile-time properties, and also adding a new compile-time property?
+[[section.usm.device.allocs]]
+==== Device allocation functions with properties support
+
+[[table.usm.device.allocs]]
+.USM Device Memory Allocation Functions with properties Support
+[width="100%",options="header",separator="@",cols="65%,35%"]
```

Comment:
This table doesn't render nicely for me.  Is that just my setup or is there something wrong with it?

Reply:
I improved the formatting of the first column that contains the function declarations.